tests/unit-tests/graphpartitioning/test_metadata.py

```diff
@@ -27,12 +27,37 @@
 from graph_dataset import *
 
 
+class UTMetadataSchema(MetadataSchema):
+    def __init__(self, filename, base_dir, init_maps=True):
+        super().__init__(filename, base_dir, init_maps=init_maps)
+
+    def init_global_nid_offsets(self):
+        super()._init_global_nid_offsets()
+
+    def init_global_eid_offsets(self):
+        super()._init_global_eid_offsets()
+
+    def init_ntype_maps(self):
+        super()._init_ntype_maps()
+
+    def init_etype_maps(self):
+        super()._init_etype_maps()
+
+    def init_etype_files(self):
+        super()._init_etype_files()
+
+    def init_ntype_feature_files(self):
+        super()._init_ntype_feature_files()
+
+    def init_etype_feature_files(self):
+        super()._init_etype_feature_files()
```

Comment:
This is not really needed right? You can just do

```
my_meta = MetadataSchema(file_name, base_dir, init_maps=False)

my_meta._init_global_nid_offsets()
```

Reply:
I was hoping to put protected methods within a wrapper class and invoke methods on the wrapper class. I was told that functions with prefix "_" are only used within the class and NOT outside the class. 

Changed the code now and these protected methods are directly invoked where ever necessary, as indicated above.

Reply:
In Python we generally we use `_name` to indicate a method that shouldn't be used externally, but in reality there's no mechanism preventing us from doing that.

In terms of unit tests, it's usually bad practice to [unit test private methods](https://softwareengineering.stackexchange.com/questions/380287/why-is-unit-testing-private-methods-considered-as-bad-practice). Wrapping the class and exposing the private methods is basically the same with an extra step.

If we wanted to follow all best practices here I'd suggest thinking carefully about which methods should be exposed to the caller ("public") and only testing those, thereby decoupling the implementation from the tests. But I think that would require quite a bit of redesigning the code here.

Reply:
I agree with your comments and reasoning. But the problem is once we decouple the code by truely exposing the only needed interface to the users then that would make unit-tests a much more involved proposition and it will become extremely difficult to test the said module effectively with meaningful test cases. Anyways we can argue all we want about this and settle on a agreeable middleground. 

I notice that you still did not approve this PR yet !

Please you still have concerns and would like to propose some changes to the code please let me know and I will get them done promptly.

Reply:
I'm going to wait for @classicsong to take a look and then we can talk about approving.